### PR TITLE
Nightly spark-tests script to follow PYSP_TEST pattern [skip ci]

### LINKED ISSUE #6403: Refactor spark-tests script to follow PYSP_TEST configs pattern
Iceberg pytests are not scaled out using xdist because they have to use SPARK_SUBMIT_FLAGS as a workaround for #6351. With this being resolved we can rewrite the config as PYSP_TEST env and run Iceberg with TEST_PARALLEL >= 2

### PR DESCRIPTION
**Pull request by pxLi**
Signed-off-by: Peixin Li <pxli@nyu.edu>

fix #6403,
remove `GNU parallel` requirement,
use PYSP_TEST_ to pass config if possible,
make all tests support parallel run

verified spark311+321 locally, will monitor status of other spark shims in nightly runs

Also will update avro test after https://github.com/NVIDIA/spark-rapids/pull/6505 merged

### CODE REVIEW

**Comment by pxLi** [comment]
convert to draft. will update after https://github.com/NVIDIA/spark-rapids/pull/6505

**Comment by pxLi** [comment]
added dockerfile changes
updated avro_test and verified
ready for review, thx

**Comment by pxLi** [comment]
build

File: jenkins/Dockerfile-blossom.integration.ubuntu [hunk 1]
```diff
@@ -40,7 +40,7 @@ RUN apt-get update -y && \
 RUN add-apt-repository ppa:deadsnakes/ppa && \
```
Comment (jlowe) [inline jenkins/Dockerfile-blossom.integration.ubuntu:40]: 2022 copyrights

File: jenkins/spark-tests.sh [hunk 2]
```diff
@@ -125,53 +125,56 @@ export PYTHONPATH="$CONDA_ROOT/lib/python$PYTHON_VER/site-packages:$PYTHONPATH"
...
+  [[ ${PARALLELISM} -gt 6 ]] && PARALLELISM=6
+  MEMORY_FRACTION=$(python -c "print(1/($PARALLELISM + 0.1))")
+
+  export TEST_PARALLEL=${PARALLELISM}
+  export PYSP_TEST_spark_rapids_memory_gpu_allocFraction=${MEMORY_FRACTION}
+  export PYSP_TEST_spark_rapids_memory_gpu_maxAllocFraction=${MEMORY_FRACTION}
+else
+  export PYSP_TEST_spark_cores_max=6
+  export PYSP_TEST_spark_executor_memory=16g
+  export TEST_PARALLEL=1
```
Comment (jlowe) [inline jenkins/spark-tests.sh:176]: Curious why 1 here?  Everywhere else uses 0 and there's a comment in run_pyspark_from_build stating 0 runs more efficiently.

Comment (gerashegalov) [reply jenkins/spark-tests.sh:176]: I would also suggest to use TEST_PARALLEL=0 for consistency

Semantically 0 and 1 are equivalent due to this logic

https://github.com/NVIDIA/spark-rapids/blob/d09bc9a7df31872d253af278388b3469ca772a05/integration_tests/run_pyspark_from_build.sh#L169-L173

https://github.com/NVIDIA/spark-rapids/blob/d09bc9a7df31872d253af278388b3469ca772a05/integration_tests/run_pyspark_from_build.sh#L277

Comment (pxLi) [reply jenkins/spark-tests.sh:176]: > Curious why 1 here?

Ah I did not realize its more efficient before, was just trying to make it non-parallel.
Updated to make it consistent

File: jenkins/Dockerfile-blossom.integration.ubuntu (hunk 1 above)
Comment (pxLi) [reply jenkins/Dockerfile-blossom.integration.ubuntu:40]: thanks! missed this one

**Comment by pxLi** [comment]
build

**Review by jlowe — APPROVED** [review]

File: integration_tests/run_pyspark_from_build.sh [hunk 3]
```diff
@@ -168,8 +168,8 @@ else
 
     if [[ ${TEST_PARALLEL} -lt 2 ]];
     then
-        # With xdist 0 and 1 are the same parallelsm but
-        # 0 is more effecient
+        # With xdist 0 and 1 are the same parallelism but
+        # 0 is more efficient
```
Comment (gerashegalov) [inline integration_tests/run_pyspark_from_build.sh:172]: nit: after we no longer start Spark on xdist master process the major difference is no longer in efficiency but how the job is submitted

**Review by gerashegalov — APPROVED** [review]
LGTM

File: integration_tests/run_pyspark_from_build.sh (hunk 3 above)
Comment (pxLi) [reply integration_tests/run_pyspark_from_build.sh:172]: Got it, thx~Will rephrase this along some other changes

**Pull request merged**